src/pru-motion-queue_test.cc

```diff
@@ -44,188 +44,122 @@ class MockPRUInterface : public PruHardwareInterface {
     return true;
   }
 
-  void SimRun(unsigned int execution_index, const uint32_t loops_left) {
-    mmap->status.index = execution_index % QUEUE_LEN;
-    mmap->status.counter = loops_left;
-    // For now we set as always empty (executed) the state of each segment
-    for (int i = 0; i < QUEUE_LEN; ++i) {
-      mmap->ring_buffer[i].state = STATE_EMPTY;
+  void SimRun(int num_exec, const uint32_t loops_left,
+              bool last_not_executed = true) {
+    // Simulate the execution of num_exec motion segments
+    for (int i=0; i < num_exec; ++i) {
+      execution_index_ = (execution_index_ + 1) % QUEUE_LEN;
+      assert(mmap->ring_buffer[execution_index_].state != STATE_EMPTY);
+      mmap->ring_buffer[execution_index_].state = STATE_EMPTY;
     }
+    if (last_not_executed || loops_left )
+      mmap->ring_buffer[execution_index_].state = STATE_FILLED;
+    mmap->status.index = execution_index_;
+    mmap->status.counter = loops_left;
   }
 
 private:
   struct MockPRUCommunication *mmap;
+  unsigned int execution_index_;
 };
 
-// Check that on init, the initial position is 0.
-TEST(RealtimePosition, init_pos) {
-  MotorsRegister absolute_pos_loops;
-  MockPRUInterface *pru_interface = new MockPRUInterface();
-  HardwareMapping *hmap = new HardwareMapping();
-  PRUMotionQueue motion_backend(hmap, (PruHardwareInterface*) pru_interface);
-
-  motion_backend.GetMotorsLoops(&absolute_pos_loops);
-
-  const MotorsRegister expected = {0, 0, 0, 0, 0, 0, 0, 0};
-  EXPECT_THAT(expected, ::testing::ContainerEq(absolute_pos_loops));
+static void check_buffer_value(PRUMotionQueue *motion_backend,
```

Comment:
Instead of this function call, I think it is better if wherever we call this function would directly call the EXPECT_EQ()

EXPECT_EQ(motion_backend.GetPendingElements(NULL), 0);

Because EXPECT_EQ() is a macro, it can log the right line number whenever this is happening (with the check_buffer_value() it would only complain about line number 69, which is less helpful)